packages/common/driver-definitions/src/storage.ts

```diff
@@ -177,6 +178,11 @@ export interface IDocumentStorageService extends Partial<IDisposable> {
      * server has deleted it this call may result in a broken promise.
      */
     downloadSummary(handle: ISummaryHandle): Promise<ISummaryTree>;
+
+    /**
+     * TODO: left as optional for now to keep implementing classes cleaner
+     */
+    getDocumentAttributesFromTree?(tree: ISnapshotTree): Promise<IDocumentAttributes>;
```

Comment:
I do not think this belongs here. From driver implementation perspective, that can be implemented as stand-alone function, so it does not belong on this interface.
If we really want to, we can always have different interface that we expose to runtime that does implement this API. But its presence is an indication of leakage - we leak non-virtualized data in the form of .attribute blob in a protocol tree. Exposing this method does not fix it (someone not knowing about this API would still get to wrong data).
See my other comments in runtime around usage on how we might approach it.